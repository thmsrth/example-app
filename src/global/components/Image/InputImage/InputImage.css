.FormInput__input.InputImage__input {
  width: calc(100% - 120px);
  display: inline-block;
}

.InputImage__upload {
  width: 120px;
  margin: calc(var(--section-margin) / 2) 0;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  display: inline-block;
  /* to align upload button to inputfield image */
  padding: calc(var(--button-space) / 2.5) calc(var(--button-space) / 2);
}

.InputImage__upload--uploaded {
  background-color: var(--color-white);
  border: none;
}

.InputImage__upload--uploaded .Button__text,
.InputImage__upload--uploaded .Button__icon {
  color: var(--color-confirm);
}

.InputImage__upload--uploaded:hover {
  cursor: default;
}

.InputImage__image {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 0 var(--section-padding);
}

.InputImage__delete {
  color: var(--action-secondary-color);
  background-color: var(--action-secondary-background-color);
  border: none;
  font-size: 1.3em;
  padding: 0 var(--section-padding);
}

.InputImage__alertMsg,
.InputImage__warningMsg,
.FormInput__label.InputImage__label__bottom {
  padding: 0 var(--section-padding);
}

.FormInput__label.InputImage__label__top {
  padding-bottom: 0;
}
