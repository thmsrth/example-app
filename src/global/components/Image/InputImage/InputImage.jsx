import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';
import I18nGlobal from '../../../utils/I18nGlobal';

import DataStateNotifier from '../../DataStateNotifier/index';
import Button from '../../Button/index';

import './InputImage.css';

const InputImage = ({
  className, name, label, value, onBlur, onChange, validationKeys,
  uploadImage, uploading, uploadError, resetImageUploadError, deleteImage,
  imageFile, format, accept, loading,
}) => {
  const {
    required,
    validationAlert,
    warningAlert,
  } = validationKeys || InputImage.defaultProps.validationKeys;

  return (
    <DataStateNotifier
      dataLoading={loading}
      dataError={uploadError}
      resetDataError={resetImageUploadError}
    >
      <div className={`InputImage ${className} ${
        bemCls(`InputImage ${className}`, `__${name}`)} ${
        bemCls(`InputImage ${className}`, '__required', required)} ${
        bemCls(`InputImage ${className}`, '--alert', validationAlert)}`}
      >
        <label
          className={`${bemCls(`InputImage ${className}`, '__label')} ${
            bemCls(`InputImage ${className}`, '__label__top')} ${
            bemCls(`InputImage ${className}`, `__${name}__label`)}`}
        >
          {`${label}${required ? '*' : ''}`}
        </label>
        <div className={`${bemCls(`InputImage ${className}`, '__alertMsg')} ${
          bemCls(`InputImage ${className}`, `__${name}__alertMsg`)}`}
        >
          {validationAlert}
        </div>
        {value
        && (
        <div className={bemCls(`InputImage ${className}`, '__image')}>
          <img
            className={`InputImage__preview Main__image--${format}`}
            alt={value}
            title={value}
            src={value}
          />
          <Button
            className={bemCls(`InputImage ${className}`, '__delete')}
            iconName="remove"
            onClick={deleteImage}
          />
        </div>
        )
        }
        <label
          className={`${bemCls(`InputImage ${className}`, '__label')} ${
            bemCls(`InputImage ${className}`, '__label__bottom')} ${
            bemCls(`InputImage ${className}`, `__${name}__label`)}`}
        >
          <input
            type="file"
            className={`${bemCls(`InputImage ${className}`, '__input')} ${
              bemCls(`InputImage ${className}`, `__${name}__input`)}`}
            name={name}
            accept={accept}
            required={required}
            onBlur={onBlur}
            onChange={onChange}
          />
          <Button
            className={`${bemCls(`InputImage ${className}`, '__upload')} ${
              bemCls(`InputImage ${className}`, '__upload--uploaded', !!value)}`}
            iconName={value ? 'check' : 'upload'}
            text={value
              ? I18nGlobal.t('en.image.states.uploaded')
              : (uploading || I18nGlobal.t('en.actions.upload'))}
            onClick={uploadImage}
            loading={!!uploading}
            disabled={!imageFile}
          />
        </label>
        <div className={`${bemCls(`InputImage ${className}`, '__warningMsg')} ${
          bemCls(`InputImage ${className}`, `__${name}__warningMsg`)}`}
        >
          {warningAlert}
        </div>
      </div>
    </DataStateNotifier>
  );
};

export default InputImage;

InputImage.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
  ]),
  validationKeys: PropTypes.shape({
    required: PropTypes.bool,
    validationAlert: PropTypes.string,
    warningAlert: PropTypes.string,
  }),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  loading: PropTypes.string,
  uploading: PropTypes.string,
  uploadError: PropTypes.string,
  imageFile: PropTypes.instanceOf(File),
  format: PropTypes.string,
  accept: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  uploadImage: PropTypes.func,
  deleteImage: PropTypes.func,
  resetImageUploadError: PropTypes.func,
};

InputImage.defaultProps = {
  className: '',
  label: '',
  value: '',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  onChange: () => {},
  onBlur: () => {},
  loading: '',
  uploading: '',
  uploadError: '',
  imageFile: null,
  format: 'xs',
  accept: false,
  uploadImage: () => {},
  deleteImage: () => {},
  resetImageUploadError: () => {},
};
