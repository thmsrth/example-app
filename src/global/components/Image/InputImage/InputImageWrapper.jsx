import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getImagePathByFormat, getImageUrlFromPath } from '../../../utils/ImageHelpers';
import SecurityUtil from '../../../security/utils/SecurityUtil';

import imageType from '../types/image';

import InputImage from './InputImage';

class InputImageWrapper extends Component {
  constructor() {
    super();
    this.handleUpload = this.handleUpload.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleFileChange = this.handleFileChange.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
    this.state = {
      imageFile: null,
    };
  }

  componentDidMount() {
    const {
      image, value, fetchImageById,
    } = this.props;
    if (!image && value) {
      fetchImageById({ id: value });
    }
  }

  handleUpload() {
    const {
      uploadImage, folderPath, renderer, name,
    } = this.props;
    const { imageFile } = this.state;

    if (imageFile) {
      const formData = new FormData();
      formData.append('file', imageFile);
      formData.set('folderPath', folderPath);
      formData.set('createdBy', SecurityUtil.getCurrentUserName());

      const uploadParams = {
        data: formData,
        onSuccess: this.handleChange,
        renderer,
        name,
      };

      uploadImage(uploadParams);
    }
  }

  handleChange(imageId) {
    const { name, onChange } = this.props;

    onChange(name, imageId);
    this.setState({ imageFile: null });
  }

  handleFileChange(event) {
    const { name, onChange } = this.props;
    const imageFile = event.target.files[0];
    this.setState({ imageFile });

    onChange(name, null);
  }

  handleDelete() {
    const { name, onChange } = this.props;
    onChange(name, null);
  }

  handleBlur() {
    const {
      name, validationKeys, onBlur,
    } = this.props;
    const options = { type: 'file' };

    onBlur(name, validationKeys, options);
  }

  render() {
    const { imageFile } = this.state;
    const {
      image, format,
    } = this.props;
    const imagePath = image ? getImageUrlFromPath(getImagePathByFormat(image, format)) : '';

    return (
      <InputImage
        {...this.props}
        value={imagePath}
        uploadImage={this.handleUpload}
        deleteImage={this.handleDelete}
        imageFile={imageFile}
        onBlur={this.handleBlur}
        onChange={this.handleFileChange}
      />
    );
  }
}

export default InputImageWrapper;

InputImageWrapper.propTypes = {
  name: PropTypes.string,
  value: PropTypes.number,
  validationKeys: PropTypes.shape({
    required: PropTypes.bool,
    pattern: PropTypes.instanceOf(RegExp),
    validationAlert: PropTypes.string,
    warningAlert: PropTypes.string,
    onValidate: PropTypes.func,
    onWarning: PropTypes.func,
  }),
  format: PropTypes.string,
  folder: PropTypes.string,
  folderPath: PropTypes.string,
  renderer: PropTypes.string,
  image: imageType,
  uploadImage: PropTypes.func,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  fetchImageById: PropTypes.func,
};

InputImageWrapper.defaultProps = {
  name: '',
  value: null,
  validationKeys: null,
  format: '',
  folder: '',
  folderPath: '/',
  renderer: '',
  image: null,
  uploadImage: () => {},
  onBlur: () => {},
  onChange: () => {},
  fetchImageById: () => {},
};
