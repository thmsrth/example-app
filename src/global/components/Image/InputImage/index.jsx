import React from 'react';
import { connect, Provider } from 'react-redux';

import store from '../store/index';

import {
  uploadImage,
  fetchImageById,
  resetImageUploadError,
} from '../actions/imageUpload';

import {
  getUploadedImages,
  getImageLoading,
  getImageUploading,
  getImageUploadError,
  getImage,
} from '../reducers/imageUpload';

import InputImageWrapper from './InputImageWrapper';

const mapStateToProps = (state, props) => ({
  uploadedImages: getUploadedImages(state),
  uploading: getImageUploading(state, props.name),
  loading: getImageLoading(state),
  uploadError: getImageUploadError(state),
  image: getImage(state, props.value),
});

const mapDispatchToProps = dispatch => ({
  uploadImage: params => dispatch(uploadImage(params)),
  fetchImageById: params => dispatch(fetchImageById(params)),
  resetImageUploadError: () => dispatch(resetImageUploadError()),
});

const ConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(InputImageWrapper);

const InputImageContainer = props => (
  <Provider store={store}>
    <ConnectedContainer {...props} />
  </Provider>
);

export default InputImageContainer;
