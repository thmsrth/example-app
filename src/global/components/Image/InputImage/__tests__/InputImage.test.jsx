import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputImage from '../InputImage';

const props = {
  className: 'MyInputImage',
  name: 'myfile',
  label: 'My Input Label',
  value: 'My File URL',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  onChange: () => {},
  uploading: '',
  uploadError: '',
};

describe('InputImage', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<InputImage {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputImage
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputImage
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputImage
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with an uploading message through the button component', () => {
    const tree = shallow(<InputImage
      {...props}
      uploading="Uploading..."
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with an error message through the DataStateNotifier component', () => {
    const tree = shallow(<InputImage
      {...props}
      uploadError="My error message"
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
