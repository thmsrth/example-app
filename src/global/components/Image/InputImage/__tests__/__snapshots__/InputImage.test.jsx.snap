// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputImage should render the component as a required input 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile InputImage__required MyInputImage__required InputImage--alert MyInputImage--alert"
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label*
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    />
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={false}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    />
  </div>
</DataStateNotifier>
`;

exports[`InputImage should render the component correctly 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile  "
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    />
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={false}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    />
  </div>
</DataStateNotifier>
`;

exports[`InputImage should render the component with a validation alert message 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile InputImage__required MyInputImage__required InputImage--alert MyInputImage--alert"
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    >
      My validation alert message
    </div>
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={false}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    />
  </div>
</DataStateNotifier>
`;

exports[`InputImage should render the component with a warning message 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile InputImage__required MyInputImage__required InputImage--alert MyInputImage--alert"
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    />
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={false}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    >
      My warning message
    </div>
  </div>
</DataStateNotifier>
`;

exports[`InputImage should render the component with an error message through the DataStateNotifier component 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError="My error message"
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile  "
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    />
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={false}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    />
  </div>
</DataStateNotifier>
`;

exports[`InputImage should render the component with an uploading message through the button component 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="InputImage MyInputImage InputImage__myfile MyInputImage__myfile  "
  >
    <label
      className="InputImage__label MyInputImage__label InputImage__label__top MyInputImage__label__top InputImage__myfile__label MyInputImage__myfile__label"
    >
      My Input Label
    </label>
    <div
      className="InputImage__alertMsg MyInputImage__alertMsg InputImage__myfile__alertMsg MyInputImage__myfile__alertMsg"
    />
    <div
      className="InputImage__image MyInputImage__image"
    >
      <img
        alt="My File URL"
        className="InputImage__preview Main__image--xs"
        src="My File URL"
        title="My File URL"
      />
      <ButtonWrapper
        className="InputImage__delete MyInputImage__delete"
        iconName="remove"
        onClick={[Function]}
        params={Object {}}
      />
    </div>
    <label
      className="InputImage__label MyInputImage__label InputImage__label__bottom MyInputImage__label__bottom InputImage__myfile__label MyInputImage__myfile__label"
    >
      <input
        accept={false}
        className="InputImage__input MyInputImage__input InputImage__myfile__input MyInputImage__myfile__input"
        name="myfile"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        type="file"
      />
      <ButtonWrapper
        className="InputImage__upload MyInputImage__upload InputImage__upload--uploaded MyInputImage__upload--uploaded"
        disabled={true}
        iconName="check"
        loading={true}
        onClick={[Function]}
        params={Object {}}
        text="Uploaded"
      />
    </label>
    <div
      className="InputImage__warningMsg MyInputImage__warningMsg InputImage__myfile__warningMsg MyInputImage__myfile__warningMsg"
    />
  </div>
</DataStateNotifier>
`;
